### PR TITLE
🌐 Add Portuguese translation for `docs/pt/docs/advanced/events.md`

### PR DESCRIPTION
**Pull request by oandersonmagalhaes**

### CODE REVIEW

File: docs/pt/docs/advanced/events.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+Vamos imaginar que o carregamento do modelo pode **levar bastante tempo**, porque ele pode ler uma grande quantidade de **dados do disco**. Então você não quer fazer isso a cada request.
+
+Você poderia carregá-lo em um nível mais alto do módulo/arquivo, mas isso poderia também significaria **carregar o modelo** mesmo se você estiver executando um simples teste automatizado, então esse teste poderia ficar **lento** por causa porque teria que esperar o carregamento do modelo antes de ser capaz de executar uma parte independete do código.
+
+
+Isso é o que nós iremos resolver, vamos carregar o modelo antes das requests serem manuseadas, mas apenas um pouco antes da aplicação iniciar o recebimento de requests, não enquanto o código estiver sendo carregado.
+
+## Lifespan
+
+Você pode definí-lo com lógica de *inicialização* e *desligamento* usando os parâmetros de `lifespan` da aplicação `FastAPI`, e um "gerenciador de contexto" (I'll show you what that is in a second).
```
Comment (fcrozetta) [inline docs/pt/docs/advanced/events.md:28]:
```suggestion
Você pode defini-lo com lógica de *inicialização* e *desligamento* usando os parâmetros de `lifespan` da aplicação `FastAPI`, e um "gerenciador de contexto" (Te mostrarei o que é isso a seguir).
```

Comment (fcrozetta) [reply docs/pt/docs/advanced/events.md:28]: The last phrase is written in first person, and I'm not sure this is a good idea, but I'm keeping it like this to match the english version

**Review by fcrozetta — CHANGES_REQUESTED** [review]
Suggestions:
- One small grammar change (My spell check keeps pointing the accent should not be there 🤷🏻‍♂️)
- the phrase in parenthesis was not translated to BR🇧🇷

**Review by fcrozetta — APPROVED** [review]

**Review by fcrozetta — APPROVED** [review]

**Comment by LorhanSohaky** [comment]
> @LorhanSohaky and @fabioueno, could you take a look at this PR, please? 😅

At the end of the day I will review

File: docs/pt/docs/advanced/events.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# Eventos de vida útil
+
+Você pode definir a lógica (código) que poderia ser executada antes da aplicação **começar**. Isso significa que esse código será executado **uma vez**, **antes** da aplicação **começar a receber requests**.
+
+Do mesmo modo, você pode definir a lógica (código) que será executada quando a aplicação estiver **desligando**. Nesse caso, esse código será executado **uma vez**, **depois** de ter possivelmente tratado **várias requests**.
+
+Por conta desse código ser executado antes da aplição **começar** a receber requests, e logo depois disso **termina** de lidar com as requests, isso cobre toda a **vida útil** da aplicação (o termo "vida útil" será importante em um segundo 😉).
+
+Pode ser muito útil para configurar **recursos** que você precisa usar por toda aplicação, e que são **compartilhadas** entre os requests, e/ou que você precisa **limpar** após. Por exemplo, o pool de uma conexão com o banco, ou carregamento de um modelo compartilhado de _machine learning_.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:9]:
```suggestion
Pode ser muito útil para configurar **recursos** que você precisa usar por toda aplicação, e que são **compartilhados** entre as requisições, e/ou que você precisa **limpar** depois. Por exemplo, o pool de uma conexão com o banco de dados ou carregamento de um modelo compartilhado de aprendizado de máquina (_machine learning_).
```

File: docs/pt/docs/advanced/events.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+## Caso de uso
+
+Vamos iniciar com um exemplo de **caso de uso** e então ver como resolvê-lo com isso.
+
+Vamos imaginar que você tem algum **modelos de _machine learning_** que você quer usar para lidar com os requests. 🤖
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:15]:
```suggestion
Vamos imaginar que você tem alguns **modelos de _machine learning_** que deseja usar para lidar com as requisições. 🤖
```

File: docs/pt/docs/advanced/events.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+Os mesmos modelos são compartilhados entre os requests, então, não é um modelo por request, ou um por usuário ou algo parecido.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:17]:
```suggestion
Os mesmos modelos são compartilhados entre as requisições, então não é um modelo por requisição, ou um por usuário ou algo parecido.
```

File: docs/pt/docs/advanced/events.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+Vamos imaginar que o carregamento do modelo pode **levar bastante tempo**, porque ele pode ler uma grande quantidade de **dados do disco**. Então você não quer fazer isso a cada request.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:19]:
```suggestion
Vamos imaginar que o carregamento do modelo pode **demorar bastante tempo**, porque ele tem que ler muitos **dados do disco**. Então você não quer fazer isso a cada requisição.
```

File: docs/pt/docs/advanced/events.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+Você poderia carregá-lo em um nível mais alto do módulo/arquivo, mas isso poderia também significaria **carregar o modelo** mesmo se você estiver executando um simples teste automatizado, então esse teste poderia ficar **lento** por causa porque teria que esperar o carregamento do modelo antes de ser capaz de executar uma parte independete do código.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:21]:
```suggestion
Você poderia carregá-lo no nível mais alto do módulo/arquivo, mas isso também poderia significaria **carregar o modelo** mesmo se você estiver executando um simples teste automatizado, então esse teste poderia ser **lento** porque teria que esperar o carregamento do modelo antes de ser capaz de executar uma parte independente do código.
```

File: docs/pt/docs/advanced/events.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+
+Isso é o que nós iremos resolver, vamos carregar o modelo antes das requests serem manuseadas, mas apenas um pouco antes da aplicação iniciar o recebimento de requests, não enquanto o código estiver sendo carregado.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:24]:
```suggestion
Isso é que nós iremos resolver, vamos carregar o modelo antes das requisições serem manuseadas, mas apenas um pouco antes da aplicação começar a receber requisições, não enquanto o código estiver sendo carregado.
```

File: docs/pt/docs/advanced/events.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+## Lifespan
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:26]:
```suggestion
## Vida útil (_Lifespan_)
```

File: docs/pt/docs/advanced/events.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+Você pode defini-lo com lógica de *inicialização* e *desligamento* usando os parâmetros de `lifespan` da aplicação `FastAPI`, e um "gerenciador de contexto" (Te mostrarei o que é isso a seguir).
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:28]:
```suggestion
Você pode definir essa lógica de *inicialização* e *encerramento* usando os parâmetros de `lifespan` da aplicação `FastAPI`, e um "gerenciador de contexto" (te mostrarei o que é isso a seguir).
```

File: docs/pt/docs/advanced/events.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+Vamos iniciar com um exemplo e ver isso detalhadamente.
+
+Nós criamos uma função assíncrona chamada `lifespan()` com `yield` como isso:
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:32]:
```suggestion
Nós criamos uma função assíncrona chamada `lifespan()` com `yield` como este:
```

File: docs/pt/docs/advanced/events.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+```Python hl_lines="16  19"
+{!../../../docs_src/events/tutorial003.py!}
+```
+
+Aqui nós estamos simulando a *inicialização* custosa de carregamento do modelo colocando a (falsa) função modelo em um dicionário em um dicinário com modelo de _machine learning_ antes do `yield`. Esse código será executado **antes** da aplicação **começar a receber requests**, durante a *inicialização*.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:38]:
```suggestion
Aqui nós estamos simulando a *inicialização* custosa do carregamento do modelo colocando a (falsa) função de modelo no dicionário com modelos de _machine learning_ antes do `yield`. Este código será executado **antes** da aplicação **começar a receber requisições**, durante a *inicialização*.
```

File: docs/pt/docs/advanced/events.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+E então, logo após o `yield`, descarregaremos o modelo. Esse código será executado **após** a aplicação **terminar de lidar com os requests**, pouco antes do *desligamento*. Isso poderia, por exemplo, liberar recursos como memória ou GPU.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:40]:
```suggestion
E então, logo após o `yield`, descarregaremos o modelo. Esse código será executado **após** a aplicação **terminar de lidar com as requisições**, pouco antes do *encerramento*. Isso poderia, por exemplo, liberar recursos como memória ou GPU.
```

File: docs/pt/docs/advanced/events.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+!!! tip
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:42]:
```suggestion
!!! tip "Dica"
```

File: docs/pt/docs/advanced/events.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+    O `shutdown` aconteceria quando você está **parando** a aplicação.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:43]:
```suggestion
    O `shutdown` aconteceria quando você estivesse **encerrando** a aplicação.
```

File: docs/pt/docs/advanced/events.md [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+
+    Talvez você precise inicializar uma nova versão, ou apenas cansou de executá-la. 🤷
+
+### Função _lifespan_
+
+A primeira coisa a notar, é que nós estamos definindo uma função assíncrona com `yield`. Isso é muito semelhante à Dependências com `yield`.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:49]:
```suggestion
A primeira coisa a notar, é que estamos definindo uma função assíncrona com `yield`. Isso é muito semelhante à Dependências com `yield`.
```

File: docs/pt/docs/advanced/events.md [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+
+```Python hl_lines="14-19"
+{!../../../docs_src/events/tutorial003.py!}
+```
+
+A primeira parte da função, antes do `yield`, irá ser executada **antes** da aplicação começar.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:55]:
```suggestion
A primeira parte da função, antes do `yield`, será  executada **antes** da aplicação inicializar.
```

File: docs/pt/docs/advanced/events.md [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+
+E a parte posterior do `yield` irá executar **após** a aplicação ser finalizada.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:57]:
```suggestion
E a parte posterior do `yield` irá executar **após** a aplicação ser encerrada.
```

File: docs/pt/docs/advanced/events.md [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+```
+
+Um **gerenciador de contexto** em Python é algo que você pode usar em uma declaração `with`, por exemplo, `open()` pode ser usado como um gerenciador de contexto:
+
+```Python
+with open("file.txt") as file:
+    file.read()
+```
+
+Nas versões mais recentes de Python, há também um **gerenciador de contexto assíncrono**. Você deveria usar isso com `async with`:
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:76]:
```suggestion
Nas versões mais recentes de Python, há também um **gerenciador de contexto assíncrono**. Você o usaria com `async with`:
```

File: docs/pt/docs/advanced/events.md [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+
+```Python
+async with lifespan(app):
+    await do_stuff()
+```
+
+Quando você cria um gerenciador de contexto ou um gerenciador de contexto assíncrono como o mencionado acima, o que ele faz é que, antes de entrar no bloco `with`, ele irá executar o código anterior ao `yield`, e depois de sair do bloco `with`, ele irá executar o código depois do `yield`.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:83]:
```suggestion
Quando você cria um gerenciador de contexto ou um gerenciador de contexto assíncrono como mencionado acima, o que ele faz é que, antes de entrar no bloco `with`, ele irá executar o código anterior ao `yield`, e depois de sair do bloco `with`, ele irá executar o código depois do `yield`.
```

File: docs/pt/docs/advanced/events.md [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+
+No nosso exemplo de código acima, nós não usamos ele diretamente, mas nós passamos para o FastAPI para ele usá-lo.
+
+O parâmetro `lifespan` da aplicação `FastAPI` leva um **Gerenciador de Contexto Assíncrono**, então nós podemos passar nosso novo gerenciador de contexto assíncrono do `lifespan` para ele.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:87]:
```suggestion
O parâmetro `lifespan` da aplicação `FastAPI` usa um **Gerenciador de Contexto Assíncrono**, então nós podemos passar nosso novo gerenciador de contexto assíncrono do `lifespan` para ele.
```

File: docs/pt/docs/advanced/events.md [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+
+```Python hl_lines="22"
+{!../../../docs_src/events/tutorial003.py!}
+```
+
+## Eventos alternativos (deprecados)
+
+!!! warning
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:95]:
```suggestion
!!! warning "Aviso"
```

File: docs/pt/docs/advanced/events.md [hunk 22]
```diff
@@ -0,0 +1,163 @@
...
+    O caminho recomendável para lidar com a *inicialização* e o *desligamento* é usando o parâmetro `lifespan` da aplicação `FastAPI` como descrito acima.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:96]:
```suggestion
    A maneira recomendada para lidar com a *inicialização* e o *encerramento* é usando o parâmetro `lifespan` da aplicação `FastAPI` como descrito acima.
```

File: docs/pt/docs/advanced/events.md [hunk 23]
```diff
@@ -0,0 +1,163 @@
...
+
+    Você provavelmente pode pular essa parte.
+
+Há um caminho alternativo para refinir a execução dessa lógica durante *inicialização* e durante *desligamento*.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:100]:
```suggestion
Existe uma forma alternativa para definir a execução dessa lógica durante *inicialização* e durante *encerramento*.
```

File: docs/pt/docs/advanced/events.md [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+
+Você pode definir manipuladores de eventos (funções) que precisam ser executadas antes da aplicação começar, ou quando a aplicação está desligando.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:102]:
```suggestion
Você pode definir manipuladores de eventos (funções) que precisam ser executadas antes da aplicação inicializar, ou quando a aplicação estiver encerrando.
```

File: docs/pt/docs/advanced/events.md [hunk 25]
```diff
@@ -0,0 +1,163 @@
...
+
+### Evento `startup`
+
+Para adicionar uma função que deve rodar antes da aplicação iniciar, declare-a com o evento `"startup"`:
+
+```Python hl_lines="8"
+{!../../../docs_src/events/tutorial001.py!}
+```
+
+Nesse caso, a função de manipulação de evento `startup` irá inicializar os itens do "database" (só um `dict`) com alguns valores.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:114]:
```suggestion
Nesse caso, a função de manipulação de evento `startup` irá inicializar os itens do "banco de dados" (só um `dict`) com alguns valores.
```

File: docs/pt/docs/advanced/events.md [hunk 26]
```diff
@@ -0,0 +1,163 @@
...
+
+Você pode adicionar mais que uma função de manipulação de evento.
+
+E sua aplicação não irá começar a receber requests até que tudo da manipulação de evento `startup` esteja completo.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:118]:
```suggestion
E sua aplicação não irá começar a receber requisições até que todos so manipuladores de eventos de `startup` sejam concluídos.
```

File: docs/pt/docs/advanced/events.md [hunk 27]
```diff
@@ -0,0 +1,163 @@
...
+
+### Evento `shutdown`
+
+Para adicionar uma função que deve ser executada quando a aplicação estiver desligando, declare ela com o evento `"shutdown"`:
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:122]:
```suggestion
Para adicionar uma função que deve ser executada quando a aplicação estiver encerrando, declare ela com o evento `"shutdown"`:
```

File: docs/pt/docs/advanced/events.md [hunk 28]
```diff
@@ -0,0 +1,163 @@
...
+
+```Python hl_lines="6"
+{!../../../docs_src/events/tutorial002.py!}
+```
+
+Aqui, a função de manipulação de evento `shutdown` irá escrever uma linha de texto `"Application shutdown"` no arquivo `log.txt`.
+
+!!! info
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:130]:
```suggestion
!!! info "Informação"
```

File: docs/pt/docs/advanced/events.md [hunk 29]
```diff
@@ -0,0 +1,163 @@
...
+    Na função `open()`, o `mode="a"` significa "acrescentar", então, a linha irá ser adicionada depois de qualquer coisa que esteja naquele arquivo, sem sobrescrever o conteúdo anterior.
+
+!!! tip
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:133]:
```suggestion
!!! tip "Dica"
```

File: docs/pt/docs/advanced/events.md [hunk 30]
```diff
@@ -0,0 +1,163 @@
...
+    Perceba que nesse caso nós estamos usando a função padrão do Python `open()` que interage com um arquivo.
+
+    Então, isso envolve I/O (input/output), que requer "espera" para coisas serem escritas em disco.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:136]:
```suggestion
    Então, isso envolve I/O (input/output), que exige"esperar" que coisas sejam escritas em disco.
```

File: docs/pt/docs/advanced/events.md [hunk 31]
```diff
@@ -0,0 +1,163 @@
...
+
+    Mas `open()` não usa `async` e `await`.
+
+    Então, nós declaramos uma função de manipulação de evento com o padrão `def` ao invés de `async def`.
+
+### `startup` e `shutdown` juntos
+
+Há uma grande chance que a lógica para sua *inicialização* e *desligamento* esteja conectada, você pode querer iniciar alguma coisa e então finalizá-la, adquirir um recurso e então liberá-lo, etc.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:144]:
```suggestion
Há uma grande chance que a lógica para sua *inicialização* e *encerramento* esteja conectada, você pode querer iniciar alguma coisa e então finalizá-la, adquirir um recurso e então liberá-lo, etc.
```

File: docs/pt/docs/advanced/events.md [hunk 32]
```diff
@@ -0,0 +1,163 @@
...
+
+Por causa disso, agora é recomendado em vez disso usar o `lifespan` como explicado acima.
+
+## Detalhes técnicos
+
+Só um detalhe técnico para nerds curiosos. 🤓
+
+Por baixo, na especificação técnica ASGI, essa é a parte do <a href="https://asgi.readthedocs.io/en/latest/specs/lifespan.html" class="external-link" target="_blank">Protocolo Lifespan</a>, e define eventos chamados `startup` e `shutdown`.
+
+!!! info
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/events.md:156]:
```suggestion
!!! info "Informação"
```

**Review by LorhanSohaky — CHANGES_REQUESTED** [review]
I fixed some typos, standardized some terms used throughout the text and
I think that with these suggestions the reading becomes more fluid

File: docs/pt/docs/advanced/events.md (hunk 26 above)
Comment (oandersonmagalhaes) [reply docs/pt/docs/advanced/events.md:118]:
```suggestion
E sua aplicação não irá começar a receber requisições até que todos os manipuladores de eventos de `startup` sejam concluídos.
```

**Review by LorhanSohaky — APPROVED** [review]
Nice!!!

**Comment by tiangolo** [comment]
Awesome, thanks @oandersonmagalhaes! :rocket: 

And thanks for the reviews @LorhanSohaky and @fcrozetta :bow:

**Pull request merged**